staging/src/k8s.io/legacy-cloud-providers/azure/azure_controller_standard.go

```diff
@@ -42,8 +42,7 @@ func (as *availabilitySet) AttachDisk(isManagedDisk bool, diskName, diskURI stri
 		return err
 	}
 
-	disks := make([]compute.DataDisk, len(*vm.StorageProfile.DataDisks))
-	copy(disks, *vm.StorageProfile.DataDisks)
+	disks := filterDetachingDisks(*vm.StorageProfile.DataDisks)
```

Comment:
if the disk is under detaching, then attach would still error, right? could we add a check here and fail earlier?

Reply:
I think this PR is on the right way, when there is disk under detaching(actually that issue is already fixed by #85158, ), this RP would remove those detaching disks and go on attaching another disk, this works according to vmss team's spec.